pkg/utils/build/quay.go

```diff
@@ -103,3 +104,35 @@ func DoesTagExistsInQuay(imageURL string) (bool, error) {
 	}
 	return false, nil
 }
+
+func IsImageRepoPublic(quayImageRepoName string) (bool, error) {
+	isPublic, err := quayClient.IsRepositoryPublic(quayOrg, quayImageRepoName)
+	if isPublic {
+		return true, nil
+	} else if !isPublic && err == nil {
+		return false, nil
+	} else {
+		return false, err
+	}
+}
```

Comment:
*quayClient.IsRepositoryPublic(quayOrg, quayImageRepoName)* already returns a bool and an error.
Can't we have just
```
isPublic, err := quayClient.IsRepositoryPublic(quayOrg, quayImageRepoName)
return isPublic,err
```

Or is this a golang no-no?

Reply:
yes we can do that as well, which will make the function simpler.

Reply:
on the other hand, do we even need this function? can't we call *quayClient.IsRepositoryPublic(quayOrg, quayImageRepoName)* directly? Since we are not changing the functionality in *IsImageRepoPublic*

Reply:
We can call it directly from the test level. But I prefer the current way of setting up quayClient and keeping all the quay related functions inside the build package with wrapper functions as well if needed, so that we have to write less code at the test level.

Reply:
I personally don't see any benefit in this case, but if you prefer it this way then it's ok to keep it.